src/main/java/jenkins/plugins/office365connector/Webhook.java

```diff
@@ -52,18 +54,23 @@
 
     private List<FactDefinition> factDefinitions = Collections.emptyList();
 
+    @Override
+    public DescriptorImpl getDescriptor() {
+            return (DescriptorImpl)super.getDescriptor();
+    }
+
     @DataBoundConstructor
     public Webhook(String url) {
-        this.url = url;
+        this.url = StringUtils.isEmpty(url) ? getDescriptor().getUrl() : url;
```

Comment:
url seems to be mandatory - what does it change?

Reply:
It covers a use case we have where we in our Jenkinsfile add a webhook to a job where the URL is not specified. We want it to be picked up from the global settings that have already been saved.

`office365ConnectorWebhooks([[
            name: "MS Teams",
            notifySuccess: true,
            notifyFailure: true,
            notifyUnstable: true,
            notifyBackToNormal: true]])`

If we don't do the change in the constructor, the resulting job will not have a URL set for the webook.